### ISSUE TITLE
[SPIR-V] Texture2D.Load with incorrect operands generates invalid spir-v instead of usable error message

### ISSUE THREAD

**Issue by MartinWickhamFB** [microsoft/DirectXShaderCompiler#6149]
**Description**
An easy newbie mistake is to call `Texture2D.Load(x, y)` instead of `Texture2D.Load(int3(x, y, 0))`.  This mistake should be caught by the type checker, and an error raised saying something like "Texture2D.Load() parameter 0 must be int3".  Instead, DXC generates invalid spir-v.  Luckily this triggers an error in the validator, rather than making it into the app and crashing in the driver.  But a better error would be nice.

**Steps to Reproduce**
```hlsl
Texture2D SourceTexture;
void Main(in float4 SvPosition : SV_Position, out float4 OutColor : SV_Target0) {
    OutColor = SourceTexture.Load(SvPosition.x, SvPosition.y);
}
```
Compile the above code with the following command line:
```
dxc -HV 2021 -T ps_6_2 -E Main -spirv -fspv-target-env=vulkan1.1 -Fo broken.spv broken.hlsl
```

[Link to godbolt repro](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:hlsl,selection:(endColumn:1,endLineNumber:5,positionColumn:1,positionLineNumber:5,selectionStartColumn:1,selectionStartLineNumber:5,startColumn:1,startLineNumber:5),source:'Texture2D+SourceTexture%3B%0Avoid+Main(in+float4+SvPosition+:+SV_Position,+out+float4+OutColor+:+SV_Target0)+%7B%0A++++OutColor+%3D+SourceTexture.Load(SvPosition.x,+SvPosition.y)%3B%0A%7D%0A'),l:'5',n:'0',o:'HLSL+source+%231',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:dxc_1_7_2207,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:hlsl,libs:!(),options:'-HV+2021+-T+ps_6_2+-E+Main+-spirv+-fspv-target-env%3Dvulkan1.1',overrides:!(),selection:(endColumn:45,endLineNumber:3,positionColumn:45,positionLineNumber:3,selectionStartColumn:45,selectionStartLineNumber:3,startColumn:45,startLineNumber:3),source:1),l:'5',n:'0',o:'+DXC+1.7.2207+(Editor+%231)',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'DXC+1.7.2207',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+DXC+1.7.2207+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

**Actual Behavior**
```
fatal error: generated SPIR-V is invalid: [VUID-StandaloneSpirv-Offset-04663] Image Operand Offset can only be used with OpImage*Gather operations
  %27 = OpImageFetch %v4float %26 %22 Lod|Offset %20 %25
```

**Environment**
- DXC version `dxcompiler.dll: 1.7(dev); dxil.dll: 1.7(101.7.2212.14)` (forked version included in Unreal 5.2.1)
- Host Operating System: Windows 10 Pro 22H2 build 19045.3803
- Also reproduces on godbolt, see link above
